app/models/post.rb

```diff
@@ -281,6 +281,21 @@ def valid_board_section
     errors.add(:section, "must be in the post's board")
   end
 
+  def valid_coauthors
+    return if self.unjoined_authors.empty?
+    return if self.user.user_ids_uninteractable.empty? && self.joined_authors.empty?
+    self.unjoined_authors.each do |unjoined|
+      next if unjoined.user_ids_uninteractable.empty?
+      joined_authors = self.joined_authors + [self.user]
+      joined_authors.each do |author|
+        unless author.can_interact_with?(unjoined)
+          errors.add(:post_author, "cannot be added")
+          break
+        end
+      end
+    end
+  end
```

Comment:
It looks like we can add two people to a post where one has blocked the other, and then if either of them tries to join the thread, this will get an error? I'm not sure if that's intentional, but I think it's the case.